templates/type.mustache

```diff
@@ -7,11 +7,10 @@
 %><%#isAtomic%><%&tsType%><%/isAtomic%><%!
 
 %><%#isObject%>{
-<%#properties%>    '<%name%>'<%^isRequired%>?<%/isRequired%>: <%>type%>;<%/properties%>}<%/isObject%><%!
+<%#properties%>    '<%name%>'<%^isRequired%>?<%/isRequired%>: <%>type%>;<%/properties%><%#hasAdditionalProperties%>
+    [key: string]: <%#additionalPropertiesType%><%>type%><%/additionalPropertiesType%>;<%/hasAdditionalProperties%>}<%/isObject%><%!
```

Comment:
use the newly introduced `hasAdditionalProperties` and `additionalPropertiesType` fields in the template